### PR TITLE
multitoken transfer and query PR#359 for `eth-bridge-integration`

### PR DESCRIPTION
**Pull request by james-chf**
This PR is equivalent to https://github.com/anoma/namada/pull/359 (i.e. issue https://github.com/anoma/namada/issues/132) but retargeting the `eth-bridge-integration` branch as a base, so that we can use it there faster, as this functionality is needed by https://github.com/anoma/namada/pull/329 (i.e. issue https://github.com/anoma/namada/issues/427)

### CODE REVIEW

**Comment by james-chf** [comment]
pls update wasm

**Comment by james-chf** [comment]
pls update wasm

**Comment by james-chf** [comment]
pls update wasm

**Comment by james-chf** [comment]
cc thanks @yito88 (I can't add you as reviewer)

File: vm_env/src/token.rs [hunk 1]
```diff
@@ -73,38 +78,101 @@ pub mod tx {
...
+            None => token::balance_key(token, src),
+        };
+        let dest_key = match &sub_prefix {
+            Some(sub_prefix) => {
+                let prefix =
+                    token::multitoken_balance_prefix(token, sub_prefix);
+                token::multitoken_balance_key(&prefix, dest)
+            }
+            None => token::balance_key(token, dest),
+        };
```
Comment (batconjurer) [inline vm_env/src/token.rs:99]:
```suggestion
        let (src_key, dest_key) = match &sub_prefix {
            Some(sub_prefix) => {
                let prefix =
                    token::multitoken_balance_prefix(token, sub_prefix);
                (token::multitoken_balance_key(&prefix, src), token::multitoken_balance_key(&prefix, dest))
            }
            None => (token::balance_key(token, src), token::balance_key(token, dest)),
        };
```

**Review by batconjurer — APPROVED** [review]
A couple of tinies, but otherwise looks good

**Comment by james-chf** [comment]
pls update wasm

**Pull request merged**